### PR TITLE
MDX Migration

### LINKED ISSUE #166: Import .mdx from other .mdx files
We need to support importing `.mdx` files from within other `.mdx` files.

This will make content creation more versatile, allowing us to re-use pieces of content across lessons. 
 

One clear example could be steps that are the same in different lessons, or even side drawers that are used in different lessons (i.e. OpenZeppelin, Solidity Fundamentals, etc.)

[Read more about this task and rewards on Dework.xyz](https://app.dework.xyz/developer-dao/frontend2web3?taskId=dafc0b4b-f084-433b-bd90-133e356d200b)

### PR DESCRIPTION
**Pull request by 7i7o**
Migrate from `next-mdx-remote` to Next's default `MDXProvider` to allow imports (& more) inside `.mdx` files. 

Closes #166 on merge

## Notice (important)

This PR moves the `.mdx` files from path `/lessons` to `/pages/lessons`

## Changes

- Remove package `next-mdx-remote`
- Add packages `@next/mdx`, `@mdx-js/loader`, `@mdx-js/react` & `remark-frontmatter`
- Modify `next.config.js` to use new config
- Add `<MDXProvider>` to `_app.tsx`
  - Create a `Components.tsx` with the list of components to pass to the `.mdx` files by `MDXProvider` 
- Remove `[slug].tsx` & `[folder]`
- Move the lessons folders with the `.mdx` files inside `pages/lessons` now that we are not using slugs
  - Modify `getting-started/index.tsx` & `lessons/index.tsx` to use new path to lessons
- Refactor Components to be used by `.mdx` files and move them into `components/mdx`
  - Modify Lessons to import another `.mdx` inside

### New Features

With this new feature, we can now re-use content across lessons.

### Example new Usage

Now, inside our content, we can now use this to import another `.mdx` content:
```
import OpenZeppelin from '../fundamentals/open_zeppelin.mdx'
```

And use it in the text like this (i.e. inside a SideDrawer):
```
<SideDrawer buttonText="More on Open Zeppelin">
  <OpenZeppelin/>
</SideDrawer>
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "dd-academy",
   "private": true,
+  "type": "module",
```
Comment (PBillingsby) [inline package.json:4]: Curious, what was the reason for changing this to a module?

**Review by PBillingsby — CHANGES_REQUESTED** [review]
Everything else looks good. Did some bug hunting and couldn't see anything out of place. All content drawers render the mdx!

File: package.json (hunk 1 above)
Comment (7i7o) [reply package.json:4]: I didn't find any other way. 
remark-frontmatter plug-in for mdx is ESM (not CommonJS anymore) so I can't use require in next.config.js
And that threw me the "cannot use import outside module" error when I changed next.config.js

If you know of a way to do that without using that type: module, please teach me! 😂
I don't have that much exp with this

File: components/mdx/SideDrawer.tsx [hunk 2]
```diff
@@ -25,14 +28,16 @@ function ContentSideDrawer(props: any) {
...
         style={{
           whiteSpace: 'normal',
           wordWrap: 'break-word',
         }}
+        rightIcon={<ArrowRightIcon />}
       >
-        {props.title}&nbsp;&nbsp;
-        <ArrowRightIcon />
+        {buttonText}
+        {/* &nbsp;&nbsp; */}
```
[the remark this answers is not in the document]
Comment (7i7o) [reply components/mdx/SideDrawer.tsx:39]: Totally! I'll do that when I'm at the computer 😊

File: package.json (hunk 1 above)
Comment (PBillingsby) [reply package.json:4]: Ok that makes sense. There are ways to do it though I’m not sure it’s entirely worth it without making it intrusive. 

Referring this for extra context: https://github.com/vercel/next.js/issues/9607

**Comment by PBillingsby** [comment]
LGTM!

**Review by PBillingsby — APPROVED** [review]

**Comment by 7i7o** [comment]
Merging!

**Pull request merged**